src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -475,6 +475,32 @@ void QgsMssqlProvider::loadFields()
   }
 }
 
+QString QgsMssqlProvider::quotedValue( const QVariant& value )
+{
+  if ( value.isNull() )
+    return "NULL";
+
+  switch ( value.type() )
+  {
+    case QVariant::Int:
+    case QVariant::LongLong:
+    case QVariant::Double:
+      return value.toString();
+
+    case QVariant::Bool:
+      return value.toBool() ? "1" : "0";
```

Comment:
It's copied from the Spatialite provider.  Admittedly, the only place it would be used for a boolean, the function is not being called.  I'll replace that later today and test.